lib/util/LazySet.js

```diff
@@ -101,13 +105,19 @@ class LazySet {
 				_set.add(item);
 			}
 		} else {
-			this._toDeepMerge.push(iterable);
-			this._needMerge = true;
-			// Avoid being too memory hungry
-			if (this._toDeepMerge.length > 100000) {
-				this._flatten();
-				if (this._toMerge.size > 100000) this._merge();
+			if (iterable instanceof LazySet) {
+				if (!iterable._isEmpty()) {
```

Comment:
```suggestion
				if (iterable._isEmpty()) return this;
```

Otherwise `if (this._toMerge.size > 100000) this._merge();` will also run for empty sets.